src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -47139,6 +47130,1576 @@ void M5(object? x)
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "x").WithLocation(32, 15),
+                // (40,15): warning CS8602: Dereference of a possibly null reference.
+                //             : x.ToString(); // 4
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "x").WithLocation(40, 15)
+                );
+        }
+
+        /// <remarks>Ported from <see cref="FlowTests.EqualsCondAccess_12"/>.</remarks>
+        [Fact]
+        public void EqualsCondAccess_12()
+        {
+            var source = @"
+class C
+{
+    int? M0(object obj) => null;
+
+    void M1(C? c, object? x, object? obj)
+    {
+        _ = (c?.M0(x = 0) == (int?)obj)
+            ? x.ToString() // 1
+            : x.ToString(); // 2
+    }
+
+    void M2(C? c, object? x, object? obj)
+    {
+        _ = (c?.M0(x = 0) == (int?)null)
+            ? x.ToString() // 3
+            : x.ToString();
+    }
+
+    void M3(C? c, object? x, object? obj)
+    {
+        _ = (c?.M0(x = 0) == null)
+            ? x.ToString() // 4
+            : x.ToString();
+    }
+}
+";
+            CreateNullableCompilation(source).VerifyDiagnostics(
+                // (9,15): warning CS8602: Dereference of a possibly null reference.
+                //             ? x.ToString() // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "x").WithLocation(9, 15),
+                // (10,15): warning CS8602: Dereference of a possibly null reference.
+                //             : x.ToString(); // 2
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "x").WithLocation(10, 15),
+                // (16,15): warning CS8602: Dereference of a possibly null reference.
+                //             ? x.ToString() // 3
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "x").WithLocation(16, 15),
+                // (23,15): warning CS8602: Dereference of a possibly null reference.
+                //             ? x.ToString() // 4
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "x").WithLocation(23, 15)
+                );
+        }
+
+        /// <remarks>Ported from <see cref="FlowTests.EqualsCondAccess_13"/>.</remarks>
+        [Fact]
+        public void EqualsCondAccess_13()
+        {
+            var source = @"
+class C
+{
+    long? M0(object obj) => null;
+    void M(C? c, object? x, int i)
+    {
+        _ = (c?.M0(x = 0) == i)
+            ? x.ToString()
+            : x.ToString(); // 1
+    }
+}
+";
+            CreateNullableCompilation(source).VerifyDiagnostics(
+                // (9,15): warning CS8602: Dereference of a possibly null reference.
+                //             : x.ToString(); // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "x").WithLocation(9, 15)
+                );
+        }
+
+        /// <remarks>Ported from <see cref="FlowTests.EqualsCondAccess_14"/>.</remarks>
+        [Fact]
+        public void EqualsCondAccess_14()
```

Comment:
If not already covered, consider adding a similar test where the body starts with `if (i is null) throw ...;`

Reply:
Done